v2/yarpcthrift/inbound.go

```diff
@@ -21,131 +21,38 @@
 package yarpcthrift
 
 import (
-	"bytes"
 	"context"
 
-	"go.uber.org/thriftrw/protocol"
 	"go.uber.org/thriftrw/wire"
 	yarpc "go.uber.org/yarpc/v2"
-	"go.uber.org/yarpc/v2/yarpcencoding"
+	"go.uber.org/yarpc/v2/yarpcerror"
 )
 
-var _ yarpc.UnaryTransportHandler = (*unaryTransportHandler)(nil)
+var _ yarpc.UnaryEncodingHandler = EncodingHandler(nil)
 
-// unaryTransportHandler wraps a Thrift Handler into a yarpc.UnaryTransportHandler.
-type unaryTransportHandler struct {
-	ThriftHandler Handler
-	Protocol      protocol.Protocol
-	Enveloping    bool
-}
-
-// Handle implements yarpc.UnaryTransportHandler.
-func (t unaryTransportHandler) Handle(ctx context.Context, req *yarpc.Request, reqBuf *yarpc.Buffer) (*yarpc.Response, *yarpc.Buffer, error) {
-	ctx, call := yarpc.NewInboundCall(ctx)
+// EncodingHandler wraps a Thrift handler into a yarpc.UnaryEncodingHandler.
+type EncodingHandler func(context.Context, wire.Value) (Response, error)
 
-	reqValue, responder, err := decodeRequest(call, req, reqBuf, t.Protocol, t.Enveloping)
-	if err != nil {
-		return nil, nil, err
+// Handle implements yarpc.UnaryEncodingHandler.
+func (e EncodingHandler) Handle(ctx context.Context, reqBody interface{}) (interface{}, error) {
```

Comment:
Can we make attach this method to a private struct like we had before? This (although minor) would keep the surface area smaller, not exposing this Handle function beyond what this package intends.

eg:
```go
type unaryEncodingHandler struct{
 h EncodingHandler
}

func ( e unaryEncodingHandler) Handle(...)(..){}
```